File: apps/server/aetherblog-service/blog-service/src/main/java/com/aetherblog/blog/entity/Category.java
package com.aetherblog.blog.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import lombok.ToString;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * 分类实体
 * 
 * @ref §6.1 - 核心表结构 (V2 增强版)
 */
@Data
@Entity
@Table(name = "categories", indexes = {
    @Index(name = "idx_categories_slug", columnList = "slug"),
    @Index(name = "idx_categories_parent", columnList = "parent_id"),
    @Index(name = "idx_categories_sort_order", columnList = "sort_order")
})
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, unique = true, length = 100)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String description;

    /**
     * 分类封面图 (V2新增)
     */
    @Column(name = "cover_image", length = 500)
    private String coverImage;

    /**
     * 分类图标
     */
    @Column(length = 100)
    private String icon;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Category parent;


    @OneToMany(mappedBy = "parent", cascade = CascadeType.ALL)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private List<Category> children = new ArrayList<>();

    @Column(name = "sort_order", nullable = false)
    private Integer sortOrder = 0;

    /**
     * 文章数量 (缓存字段)
     */
    @Column(name = "post_count", nullable = false)
    private Integer postCount = 0;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: apps/server-go/internal/model/category.go
package model

import "time"

// Category 对应数据库 `categories` 表，表示博客文章的分类目录。
// 支持通过 ParentID 构建多级父子分类结构。
type Category struct {
	ID          int64      `db:"id"`           // 分类主键 ID
	Name        string     `db:"name"`         // 分类显示名称
	Slug        string     `db:"slug"`         // URL 友好的唯一标识符
	Description *string    `db:"description"`  // 分类描述，可为空
	CoverImage  *string    `db:"cover_image"`  // 分类封面图片 URL，可为空
	Icon        *string    `db:"icon"`         // 分类图标标识符或 emoji，可为空
	ParentID    *int64     `db:"parent_id"`    // 父分类 ID，nil 表示顶级分类
	SortOrder   int        `db:"sort_order"`   // 显示排序权重，值越小越靠前
	PostCount   int        `db:"post_count"`   // 该分类下的文章数量（缓存值）
	CreatedAt   time.Time  `db:"created_at"`   // 记录创建时间
	UpdatedAt   time.Time  `db:"updated_at"`   // 记录最后更新时间
}
